web/src/conversation_participants.ts

```diff
@@ -0,0 +1,33 @@
+import type {Message} from "./message_store.ts";
+import * as people from "./people.ts";
+
+export class ConversationParticipants {
+    humans: Set<number>;
+    bots: Set<number>;
+
+    constructor(messages: Message[]) {
+        this.humans = new Set();
+        this.bots = new Set();
+        this.add_from_messages(messages);
+    }
```

Comment:
Let's add a comment here noting that we don't support removal of a message due to deletion or message moves, because we aren't tracking the set of message IDs for each participant, so we currently just rebuild.

(We could make a slightly more complex data structure and handle that perfectly, by just maintaining a map from user ID to a set of message IDs sent by that user in this conversation, etc., but it's considerably more work and not obviously necessary for our use cases.)

Reply:
Done!